app/src/main/java/com/endava/flights/datasource/FlightImpl.kt

```diff
@@ -0,0 +1,21 @@
+package com.endava.flights.datasource
+
+import android.text.BoringLayout
+import android.util.Log
+import com.endava.flights.model.Flight
+import com.endava.flights.model.Service
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+class FlightImpl(
```

Comment:
Here, do you think it is necessary to implement another FlightsRepository? (You are doing it to set a new dispatcher but think if is really useful if you can control the dispatcher in the UseCase) 

If you still want to maintain it, take into account the repo shouldn't be initialized in the function. Think where would be better place to init it.